include/fastdds/dds/xtypes/type_representation/TypeObject.idl

```diff
@@ -0,0 +1,1080 @@
+
+@extensibility(APPENDABLE) @nested
+struct MinimalAliasHeader {
+    // Empty. Available for future extension
+};
+
+@extensibility(FINAL) @nested
+struct CompleteAliasType {
+    AliasTypeFlag                               alias_flags;
+    CompleteAliasHeader                         header;
+    CompleteAliasBody                           body;
+};
+
+@extensibility(FINAL) @nested
+struct MinimalAliasType {
+    AliasTypeFlag                               alias_flags;
+    MinimalAliasHeader                          header;
+    MinimalAliasBody                            body;
+};
+
+// --- Collections: ----------------------------------------------------
+@extensibility(FINAL) @nested
+struct CompleteElementDetail {
+    @optional AppliedBuiltinMemberAnnotations   ann_builtin;
+    @optional AppliedAnnotationSeq              ann_custom;
+};
+
+@extensibility(FINAL) @nested
+struct CommonCollectionElement {
+    CollectionElementFlag                       element_flags;
+    TypeIdentifier                              type;
+};
+
+@extensibility(APPENDABLE) @nested
+struct CompleteCollectionElement {
+    CommonCollectionElement                     common;
+    CompleteElementDetail                       detail;
+};
+
+@extensibility(APPENDABLE) @nested
+struct MinimalCollectionElement {
+    CommonCollectionElement                     common;
+};
+
+@extensibility(FINAL) @nested
+struct CommonCollectionHeader {
+    LBound                                      bound;
+};
+
+@extensibility(APPENDABLE) @nested
+struct CompleteCollectionHeader {
+    CommonCollectionHeader                      common;
+    @optional CompleteTypeDetail                detail; // not present for anonymous
+};
+
+@extensibility(APPENDABLE) @nested
+struct MinimalCollectionHeader {
+    CommonCollectionHeader                      common;
+};
+
+// --- Sequence: ------------------------------------------------------
+@extensibility(FINAL) @nested
+struct CompleteSequenceType {
+    CollectionTypeFlag                          collection_flag;
+    CompleteCollectionHeader                    header;
+    CompleteCollectionElement                   element;
+};
+
+@extensibility(FINAL) @nested
+struct MinimalSequenceType {
+    CollectionTypeFlag                          collection_flag;
+    MinimalCollectionHeader                     header;
+    MinimalCollectionElement                    element;
+};
+
+// --- Array: ------------------------------------------------------
+@extensibility(FINAL) @nested
+struct CommonArrayHeader {
+    LBoundSeq                                   bound_seq;
+};
+
+@extensibility(APPENDABLE) @nested
+struct CompleteArrayHeader {
+    CommonArrayHeader                           common;
+    CompleteTypeDetail                          detail;
+};
+
+@extensibility(APPENDABLE) @nested
+struct MinimalArrayHeader {
+    CommonArrayHeader                           common;
+};
+
+@extensibility(APPENDABLE) @nested
+struct CompleteArrayType
+{
```

Comment:
I am not going to apply any suggested change to the IDL file as it seems that we are going to directly include the IDL coming from the specification and that Fast DDS-Gen would take charge of modifying the namespaces accordingly instead of modifying the IDL.